sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java

```diff
@@ -397,6 +405,44 @@ public void queryMetrics(String query, Boolean qmEnabled) {
         }
     }
 
+    @Test(groups = {"simple"}, timeOut = TIMEOUT)
+    public void queryDiagnosticsOnOrderBy() {
+        //  create container with more than 4 physical partitions
+        String containerId = "testcontainer";
+        cosmosAsyncDatabase.createContainer(containerId, "/mypk",
+            ThroughputProperties.createManualThroughput(40000)).block();
+        CosmosAsyncContainer testcontainer = cosmosAsyncDatabase.getContainer(containerId);
+        CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
+        testcontainer.createItem(getInternalObjectNode()).block();
+        options.setMaxDegreeOfParallelism(-1);
+        String query = "SELECT * from c ORDER BY c._ts DESC";
+        CosmosPagedFlux<InternalObjectNode> cosmosPagedFlux = testcontainer.queryItems(query, options,
+            InternalObjectNode.class);
+        AtomicInteger counterPkRid = new AtomicInteger();
+        AtomicInteger counterPartitionKeyRangeId = new AtomicInteger();
+        cosmosPagedFlux.byPage().flatMap(feedResponse -> {
+            String cosmosDiagnosticsString = feedResponse.getCosmosDiagnostics().toString();
+            //  find all partition key range ids in cosmos diagnostics
+            Pattern pattern = Pattern.compile("\"partitionKeyRangeId\":\"");
+            Matcher matcher = pattern.matcher(cosmosDiagnosticsString);
+            while (matcher.find()) {
+                counterPartitionKeyRangeId.incrementAndGet();
+            }
+            //  find all partition key range ids in query metrics
+            pattern = Pattern.compile("pkrId:");
+            matcher = pattern.matcher(cosmosDiagnosticsString);
+            while (matcher.find()) {
+                counterPkRid.incrementAndGet();
+            }
+            return Flux.just(feedResponse);
+        }).blockLast();
+
+        // assert that cosmos diagnostics has diagnostics information for all partitions same as query metrics
+        assertThat(counterPkRid.get() * 2).isEqualTo(counterPartitionKeyRangeId.get());
```

Comment:
why we need to * 2 here?

Reply:
Good catch, didn't realize, I was using Strong consistency. Will fix it.